src/api/models/lines.js

```diff
@@ -52,16 +52,79 @@ lineSchema.methods.logEvent = function() {
 // update status of line
 // used for manually updating status, instead of using api route
 // also creates log event.
+// returns 'success' upon success, and an error message otherwise.
 lineSchema.methods.updateStatus = function(status) {
-    console.log("Line: Manually Updating Status: " + status);
+    var msg = null;
+    //verify that line progression is happening in order
+    switch (status) {
```

Comment:
you could potentially do a generic solution by creating a dictionary/map to handle multiple previous states.
for example: 
```
inverseMap = {
    'notification': ['preline'],
    'inline': ['notification'],
    'startrecruiter':['inline'],
    'finishrecruiter':['startrecruiter'],
    'timeoutchurn':['preline', 'notification', 'inline'],
    'voluntarychurn':[...] // idk how u handle the all case though
    'preline': [] 
};

```

OTHERWISE see other comment below

Reply:
u don't have to do this, switch case is fine though

Reply:
in light of the backlog i'll just leave this as is. I did change it to allow inline/startrecruiter -> timeoutchurn though.